airflow/providers/common/io/xcom/backend.py

```diff
@@ -142,7 +142,12 @@ def serialize_value(
 
         base_path = _get_base_path()
         while True:  # Safeguard against collisions.
-            p = base_path.joinpath(dag_id, run_id, task_id, f"{uuid.uuid4()}{suffix}")
+            p = base_path.joinpath(
+                dag_id or "NO_DAG_ID",
```

Comment:
I was hoping you know a reason why the fileds can be None - I think actually the interface definition is "broken by design" because we never store XCom w/o DAG ID, Run ID or Task ID. I first thought making an assert but was not sure if this is a hidden feature :-D

Reply:
I prefer to fix it this way - unless someone wills to spend time on going deeper. Generally speaking - I think it comes from the "imperfectness" of Python typing system and the fact that really at this moment the fields cannot be None - because they are guaranteed to be set before, but I do not want to spend time on going through all the paths to be honest - and prefer to get a "defensive" approach - which will work and signal problems in case something like that happens. 

But if you or someone else would like to investigate - I have nothing against it.